### PR TITLE
Bug 1266756 - Fixing GBI

### PR DESCRIPTION
**Pull request by lissyx**

### CODE REVIEW

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=493adb343aae719a56441f37f4a8a7682af3bc86

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=493adb343aae719a56441f37f4a8a7682af3bc86

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=493adb343aae719a56441f37f4a8a7682af3bc86

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1d9e68d4880b472a64b2f90de44c868f140671c4

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a6acc00396b5e3e39ef03b350e5a4686f1e1c774

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8e81a4878f2437163c5a7e368430a676dd243079

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d160e29fce8f9ba2f454c11017166f7d0be2faf1

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9b54b4e429617caae78edda94dd8968a15040007

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ebbf53742f922e06fcebdf3546020a291bbc0c4f

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=26181fffae82606fc32ea41b8773bd2911098969

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e6470db27a03d691729106cd64313af089fc6902

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=a0b94b29bb5642c1b354bce5912f0ccd4f163e1e

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b092a78b8d2a9e4750b15aff5eb865e2f55418f2

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=109cdb820dc75b3a412d12de9b3ae9ad62028e64

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=bba77baf0be575378d2eee5aaf7d5d83800e02b6

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0f49ebb0ce05dfbc97a3b7905cea874c0af31ea6

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e36785a6ce57c05f0dffdf93d7dba609da889512

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0ac9a388bd6afd27c7872ab9279fb00dcefc0ce8

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1929aee1c5020c55acafb724284c374a52047339

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=faec99995eb720e5b7997a3d6ff3aa98c938ce29

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=7a104b09165da2a19357d1837f42f8cf5da22a13

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0c1d9734fb01b359577d9226c8b0ce83a16da1aa

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=001affc9cf2211b419e30d4fbdb8d3c250c349b9

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f6c1f00f6d06a38929e3057da960c69733e48ac0

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1271fc14145b6f07d0a8f11654e4d96cfaf5fe00

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=08bac4bc10a92b66f9e7bab336572d45e4e4143f

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=7ce467ab5963734032a941a55bb29cc8bfe7f8d3

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3bf1c809fbb9dd05d4c681a1c2147644901b44de

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6124c9340d16fa93b615a8eac4c6912ee610c7c9

File: build/test/integration/debug_test.js [hunk 1]
```diff
@@ -72,8 +71,9 @@ suite('Make with DEBUG=1', function() {
       helper.checkSettings(settings, expectedSettings);
       helper.checkPrefs(sandbox.userPrefs, expectedUserPrefs);
       // expect one zip file for each "external" app.
-      assert.equal(zipCount, 6, 'we should have six zip files in ' +
-        'profile-debug directory');
+      // No moore external app... now
+      // assert.equal(zipCount, 6, 'we should have six zip files in ' +
+      //   'profile-debug directory');
```
Comment (nullaus) [inline build/test/integration/debug_test.js]: Let's remove this completely.

File: build/test/integration/distribution_test.js [hunk 2]
```diff
@@ -3,7 +3,7 @@
 /* jshint maxlen: 130 */
 
 var rmrf = require('rimraf').sync;
-var AdmZip = require('adm-zip');
+// var AdmZip = require('adm-zip');
```
Comment (nullaus) [inline build/test/integration/distribution_test.js]: Might as well just delete the line. Guessing you were probably going to do that. :)

File: build/test/integration/distribution_test.js [hunk 3]
```diff
@@ -150,15 +151,19 @@ suite('Distribution mechanism', function() {
       path.join(cusDir, 'mobizilla',
         'mobizilla_expected_bookmarks.json'));
   }
+  */
```
Comment (nullaus) [inline build/test/integration/distribution_test.js]: Let's go ahead and delete this.

File: build/test/integration/distribution_test.js [hunk 4]
```diff
@@ -150,15 +151,19 @@ suite('Distribution mechanism', function() {
...
 
+  /*
   function validateWappush() {
     var wappushZipPath = path.join(process.cwd(), 'profile',
       'webapps', 'wappush.gaiamobile.org', 'application.zip');
 
     helper.checkFileContentByPathInZip(wappushZipPath, 'js/whitelist.json',
       path.join(cusDir, 'wappush-whitelist.json'), true);
   }
+  */
```
Comment (nullaus) [inline build/test/integration/distribution_test.js]: Ditto.

File: build/test/integration/distribution_test.js [hunk 5]
```diff
@@ -169,24 +174,25 @@ suite('Distribution mechanism', function() {
       'resources/customize.png',
       path.join(cusDir, 'wallpapers', 'customize.png'), false);
   }
+  */
```
Comment (nullaus) [inline build/test/integration/distribution_test.js]: And this one. :)

File: build/test/integration/distribution_test.js [hunk 6]
```diff
@@ -251,7 +258,9 @@ suite('Distribution mechanism', function() {
     }
     return expectedContent;
   }
+  */
```
Comment (nullaus) [inline build/test/integration/distribution_test.js]: delete

File: build/test/integration/distribution_test.js [hunk 7]
```diff
@@ -262,12 +271,18 @@ suite('Distribution mechanism', function() {
     var expectContent = parseCustimizeImageSetting(cusConfig);
     assert.equal(presetsContent,  expectContent);
   }
+  */
```
Comment (nullaus) [inline build/test/integration/distribution_test.js]: delete

File: build/test/integration/distribution_test.js [hunk 8]
```diff
@@ -301,12 +317,12 @@ suite('Distribution mechanism', function() {
...
-      validateOperatorVariant();
-      validateWappush();
+      // validateOperatorVariant(); // We don't have this anymore
+      // validateWappush(); // We don't have this anymore
       validateSystem();
-      validateGallery();
+      // validateGallery(); // We don't have this anymore
       validateHomescreen();
-      validateWallpaper();
+      // validateWallpaper();
```
Comment (nullaus) [inline build/test/integration/distribution_test.js]: delete

File: build/test/integration/distribution_test.js [hunk 9]
```diff
@@ -301,12 +317,12 @@ suite('Distribution mechanism', function() {
       helper.checkError(error, stdout, stderr);
       validatePreloadSettingDB();
       validateSettings();
-      validateOperatorVariant();
-      validateWappush();
+      // validateOperatorVariant(); // We don't have this anymore
+      // validateWappush(); // We don't have this anymore
```
Comment (nullaus) [inline build/test/integration/distribution_test.js]: delete two above.

File: build/test/integration/distribution_test.js [hunk 10]
```diff
@@ -301,12 +317,12 @@ suite('Distribution mechanism', function() {
...
       validateSystem();
-      validateGallery();
+      // validateGallery(); // We don't have this anymore
```
Comment (nullaus) [inline build/test/integration/distribution_test.js]: delete

File: build/test/integration/distribution_test.js [hunk 11]
```diff
@@ -281,6 +296,7 @@ suite('Distribution mechanism', function() {
     assert.isTrue(fs.existsSync(path.join(process.cwd(), 'profile',
       'svoperapps', 'Twitter', 'manifest.webapp')),
       'manifest for Twitter should exist');
+    */
   }
```
Comment (nullaus) [inline build/test/integration/distribution_test.js]: Might be good to file a bug to re-enable this later.

File: build/test/integration/l10n_test.js [hunk 12]
```diff
@@ -3,7 +3,7 @@
 var assert = require('chai').assert;
 var fs = require('fs');
 var path = require('path');
-var AdmZip = require('adm-zip');
+// var AdmZip = require('adm-zip');
```
Comment (nullaus) [inline build/test/integration/l10n_test.js]: delete

File: build/test/integration/multilocale_test.js [hunk 13]
```diff
@@ -2,22 +2,22 @@
...
 var assert = require('chai').assert;
 var path = require('path');
-var AdmZip = require('adm-zip');
+var fs = require('fs');
 var helper = require('./helper');
 
 suite('Multilocale integration tests', function() {
   var localesDir = 'build/test/resources/locales';
-  var localesFileObj = {'en-US': '', 'zh-CN': ''};
+  // var localesFileObj = {'en-US': '', 'zh-CN': ''};
```
Comment (nullaus) [inline build/test/integration/multilocale_test.js]: delete

File: build/test/integration/multilocale_test.js [hunk 14]
```diff
@@ -2,22 +2,22 @@
...
   function makeHelper(localesFilePath, localesDir, concat, done) {
-    var settingsZipPath = path.join(process.cwd(), 'profile', 'webapps',
-      'settings.gaiamobile.org', 'application.zip');
-    var cnPathInZip = 'locales-obj/index.zh-CN.json';
+    var settingsFolderPath = path.join(process.cwd(), 'profile', 'apps',
+      'settings');
+    var cnPathInFolder = 'locales-obj/index.zh-CN.json';
     var cnSettingsProperties = 'locales/settings.zh-CN.properties';
-    var langPathInZip = 'shared/resources/languages.json';
+    // var langPathInFolder = 'shared/resources/languages.json';
```
Comment (nullaus) [inline build/test/integration/multilocale_test.js]: delete

File: build/test/integration/distribution_test.js (hunk 4 above)
Comment (lissyx) [inline build/test/integration/distribution_test.js]: WapPush will be needed (we checked this on irc yesterday) so I think I will keep it :)

File: build/test/integration/multilocale_test.js (hunk 14 above)
Comment (lissyx) [inline build/test/integration/multilocale_test.js]: I prefer to keep them since they are needed to reenable the assert I disabled because of bug 1268477

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=3b49fd92c5d936e5accf2abca4b3a150d5ded766

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=31ae713e81cada6620d3f32c0c46aeaa1c0f864c

**Pull request merged**